sys/net/link_layer/lorawan/Kconfig

```diff
@@ -0,0 +1,491 @@
+if LORAMAC_REGION_US_915
+
+config LORAMAC_DEFAULT_DR_8
+    bool "DR8"
+
+config LORAMAC_DEFAULT_DR_9
+    bool "DR9"
+
+config LORAMAC_DEFAULT_DR_10
+    bool "DR10"
+
+config LORAMAC_DEFAULT_DR_11
+    bool "DR11"
+
+config LORAMAC_DEFAULT_DR_12
+    bool "DR12"
+
+config LORAMAC_DEFAULT_DR_13
+    bool "DR13"
+
+config LORAMAC_DEFAULT_DR_14
+    bool "DR14"
+
+config LORAMAC_DEFAULT_DR_15
+    bool "DR15"
+
+endif # LORAMAC_REGION_US_915
+
+endchoice
+
+choice
+    bool "TX Power index"
+    depends on USEMODULE_RN2XX3
+    default LORAMAC_DEFAULT_TX_POWER_1
+    help
+        Configure the TX power index. TXPower index indicates power levels
+        relative to the max EIRP level of the end-device. Refer
+        LoRaWAN 1.0.3 Regional Parameters for more information.
+
+config LORAMAC_DEFAULT_TX_POWER_0
+    bool "TXPower 0"
+
+config LORAMAC_DEFAULT_TX_POWER_1
+    bool "TXPower 1"
+
+config LORAMAC_DEFAULT_TX_POWER_2
+    bool "TXPower 2"
+
+config LORAMAC_DEFAULT_TX_POWER_3
+    bool "TXPower 3"
+
+config LORAMAC_DEFAULT_TX_POWER_4
+    bool "TXPower 4"
+
+config LORAMAC_DEFAULT_TX_POWER_5
+    bool "TXPower 5"
+
+config LORAMAC_DEFAULT_TX_POWER_6
+    bool "TXPower 6"
+
+config LORAMAC_DEFAULT_TX_POWER_7
+    bool "TXPower 7"
+
+if LORAMAC_REGION_US_915
+
+config LORAMAC_DEFAULT_TX_POWER_8
+    bool "TXPower 8"
+
+config LORAMAC_DEFAULT_TX_POWER_9
+    bool "TXPower 9"
+
+config LORAMAC_DEFAULT_TX_POWER_10
+    bool "TXPower 10"
+
+config LORAMAC_DEFAULT_TX_POWER_11
+    bool "TXPower 11"
+
+config LORAMAC_DEFAULT_TX_POWER_12
+    bool "TXPower 12"
+
+config LORAMAC_DEFAULT_TX_POWER_13
+    bool "TXPower 13"
+
+config LORAMAC_DEFAULT_TX_POWER_14
+    bool "TXPower 14"
+
+config LORAMAC_DEFAULT_TX_POWER_15
+    bool "TXPower 15"
+
+endif # LORAMAC_REGION_US_915
+
+endchoice
+
+config LORAMAC_DEFAULT_TX_PORT
+    int "MAC TX port"
+    default 2
+    range 1 223
+    help
+        Configure the default MAC TX port. The TX port denotes the port field
+        `FPort` in the payload and is commonly used on the destination
+        side (Application Server) to distinguish different payload formats. For
+        example it can be used to identify different sensor types/values from
+        the same end-node and hence different payload formats can be assigned to
+        the different values based on value of port.
+
+choice
+    bool "MAC TX mode"
+    default LORAMAC_DEFAULT_TX_MODE_CNF
+    help
+        Configure the type of uplink packet (confirmed/unconfirmed). A confirmed
+        uplink will expect an acknowledgment from the network server; otherwise,
+        the message will be retransmitted by the number indicated by
+        CONFIG_LORAMAC_DEFAULT_RETX, whereas an unconfirmed message will not
+        expect any acknowledgment back from the network server.
+
+config LORAMAC_DEFAULT_TX_MODE_CNF
+    bool "Confirmable"
+
+config LORAMAC_DEFAULT_TX_MODE_UNCNF
+    bool "Unconfirmable"
+
+endchoice
+
+config LORAMAC_ADVANCED_OPTIONS
```

Comment:
I'd say that this switch just hides the prompts, instead of making it a dependency for the symbols.

Reply:
Sure. Shall add this dependency to all.